src/modules/new-request-form/fields/LookupField.tsx

```diff
@@ -0,0 +1,252 @@
+import type { IComboboxProps } from "@zendeskgarden/react-dropdowns.next";
+import {
+  Field as GardenField,
+  Label,
+  Hint,
+  Combobox,
+  Option,
+  Message,
+} from "@zendeskgarden/react-dropdowns.next";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import type { Field, FieldOption } from "../data-types";
+import { Span } from "@zendeskgarden/react-typography";
+import debounce from "lodash.debounce";
+import { useTranslation } from "react-i18next";
+import { EmptyValueOption } from "./EmptyValueOption";
+
+function getCustomObjectKey(targetType: string) {
+  return targetType.replace("zen:custom_object:", "");
+}
+
+const EMPTY_OPTION = {
+  value: "",
+  name: "-",
+};
+
+interface LookupFieldProps {
+  field: Field;
+  userId: number;
+  organizationId: string | null;
+  onChange: (value: string) => void;
+}
+
+export function LookupField({
+  field,
+  userId,
+  organizationId,
+  onChange,
+}: LookupFieldProps) {
+  const {
+    id: fieldId,
+    label,
+    error,
+    value,
+    name,
+    required,
+    description,
+    relationship_target_type,
+  } = field;
+  const [options, setOptions] = useState<FieldOption[]>([]);
+  const [selectedOption, setSelectedOption] = useState<FieldOption | null>(
+    null
+  );
+  const [inputValue, setInputValue] = useState<string>(value as string);
+  const [isLoadingOptions, setIsLoadingOptions] = useState<boolean>(false);
+  const { t } = useTranslation();
+
+  const customObjectKey = getCustomObjectKey(
+    relationship_target_type as string
+  );
+
+  const loadingOption = {
+    name: t(
+      "new-request-form.lookup-field.loading-options",
+      "Loading items..."
+    ),
+    id: "loading",
+  };
+
+  const noResultsOption = {
+    name: t(
+      "new-request-form.lookup-field.no-matches-found",
+      "No matches found"
+    ),
+    id: "no-results",
+  };
+
+  const fetchSelectedOption = useCallback(
+    async (selectionValue: string) => {
+      try {
+        const res = await fetch(
+          `/api/v2/custom_objects/${customObjectKey}/records/${selectionValue}`
+        );
+        if (res.ok) {
+          const { custom_object_record } = await res.json();
+          const newSelectedOption = {
+            name: custom_object_record.name,
+            value: custom_object_record.id,
+          };
+          setSelectedOption(newSelectedOption);
+          setInputValue(custom_object_record.name);
+        }
+      } catch (error) {
+        console.log(error);
```

Comment:
Do we usually leave `console.log` calls in the code for frontend?

Reply:
Usually linters don't allow this, here linter tells me that every path needs to return value so I need to do something with this error and we don't have any logging tool added. I will change it to console.error, I see there is other usage of this in Copenhagen Theme repo.

Reply:
Thank you for checking this out. If `console.log` is the best approach for it I believe it is fine to keep. Just wanted to be sure how we usually do it.